webapp/graphite/render/datalib.py

```diff
@@ -271,6 +272,13 @@ def prefetchData(requestContext, pathExpressions):
       ),
     ))
 
+  # Several third-party readers including rrdtool and biggraphite return values in a
+  # generator which can only be iterated on once. These must be converted to a list.
+  for pathExpression, items in prefetched.items():
+    for i, (name, (time_info, values)) in enumerate(items):
+      if isinstance(result['values'], types.GeneratorType):
```

Comment:
where is `result['values']` coming from? assuming this should be `values`

Reply:
You are correct, that was a mistake introduced when I moved the check into its own loop to address an earlier change request.  Sorry about that.  I've updated the pr again.